### ISSUE TITLE
GCS name format inconsistency

### ISSUE THREAD

**Issue by ngorchakova** [OpenLineage/OpenLineage#2355]
According to the OL documentation https://github.com/OpenLineage/OpenLineage/blob/main/spec/Naming.md#gcs
the namespace for GCS should be:

- **Namespace**: gs://{bucket name}
- **Name**: Path

While the example here:
https://github.com/OpenLineage/OpenLineage/blob/108b3c077c5a9beff52115f29a789a8a3e485886/integration/spark/app/integrations/sparkrdd/2.json#L24-L25 and local tests running GCS to GCS action) give an incorrect namespace and name. 

- **namespace** = gs.{bucket name}
- **name** = "gs://{bucket name}/{path}"

Note: In case of GCS to BigQuery the namespace / name for GCS object is generated correctly.

**Comment by pawel-big-lebowski** [comment]
This was some old test that did not touch `gs` in any way,  and dataset name was not verified. 
The test relied on `MatchesMapRecursively` test helper class, which had a bug. 
I rewrote it to manual assertions testing each fields.

https://github.com/OpenLineage/OpenLineage/pull/2356

**Comment by ngorchakova** [comment]
Hi Pawel,
Thanks a lot for your prompt reply. 
It is easy to reproduce this issue by running a simple spark job:

```
from pyspark.sql import SparkSession
import sys

spark = SparkSession \
  .builder \
  .appName('LINEAGE_GCS_TO_GCS') \
  .getOrCreate()

lines = spark.sparkContext.textFile('gs://{BUCKET}/test.csv')
lines.saveAsTextFile('gs://{BUCKET}/test_output.csv')
```

The lineage event i got contains

```

"inputs": [
    {
      "namespace": "gs.{BUCKET}",
      "name": "gs://{BUCKET}/test.csv"
    }
  ],
  "outputs": [
    {
      "namespace": "gs.{BUCKET}",
      "name": "gs://{BUCKET}/test_output.csv"
    }
  ],
```

The output is exactly what you had in old test, so I assumed it is connected. 

Please note, as soon as I change the code to read the file as a CSV, the namespace and name are according to the naming convention ( https://github.com/OpenLineage/OpenLineage/blob/main/spec/Naming.md#gcs)

**Comment by pawel-big-lebowski** [comment]
What OL version are you using? We had similar issue in the past and it got fixed in [1.5.0](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md#150---2023-11-01).

I created a PR #2362 which refactors google cloud integration test to a single test suite. I also created a new test that does exactly what you describe and the test is passing.
